tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -850,5 +850,589 @@ int main(int argc, char **argv)
         free(rsa_private_key_pem);
         EXPECT_SUCCESS(s2n_connection_free(conn));
     }
+
+    /* Test chacha20 boosting behaviour */
+    {
+        /* Setup cipher preferences + security policy */
+        struct s2n_cipher_preferences cipher_preferences = { 0 };
+        struct s2n_security_policy security_policy = {
+            .minimum_protocol_version = S2N_SSLv2,
+            .cipher_preferences = &cipher_preferences,
+            .kem_preferences = &kem_preferences_null,
+            .signature_preferences = &s2n_signature_preferences_20201021,
+            .ecc_preferences = &s2n_ecc_preferences_test_all,
+        };
+
+        /* Initialise config and relevant certs */
+        DEFER_CLEANUP(struct s2n_config *server_config = s2n_config_new(), s2n_config_ptr_free);
+        struct s2n_cert_chain_and_key* rsa_cert = s2n_cert_chain_and_key_new();
+        struct s2n_cert_chain_and_key* ecdsa_cert = s2n_cert_chain_and_key_new();
+        char *rsa_cert_chain_pem = malloc(S2N_MAX_TEST_PEM_SIZE);
+        char *rsa_private_key_pem = malloc(S2N_MAX_TEST_PEM_SIZE); 
+        char *ecdsa_cert_chain_pem = malloc(S2N_MAX_TEST_PEM_SIZE);
+        char *ecdsa_private_key_pem = malloc(S2N_MAX_TEST_PEM_SIZE);
+
+        EXPECT_NOT_NULL(rsa_cert);
+        EXPECT_NOT_NULL(ecdsa_cert);
+        EXPECT_NOT_NULL(server_config);
+        EXPECT_NOT_NULL(rsa_cert_chain_pem);
+        EXPECT_NOT_NULL(rsa_private_key_pem);
+        EXPECT_NOT_NULL(ecdsa_cert_chain_pem);
+        EXPECT_NOT_NULL(ecdsa_private_key_pem);
+        EXPECT_SUCCESS(s2n_read_test_pem(S2N_DEFAULT_TEST_CERT_CHAIN, rsa_cert_chain_pem, S2N_MAX_TEST_PEM_SIZE));
+        EXPECT_SUCCESS(s2n_read_test_pem(S2N_DEFAULT_TEST_PRIVATE_KEY, rsa_private_key_pem, S2N_MAX_TEST_PEM_SIZE));
+        EXPECT_SUCCESS(s2n_read_test_pem(S2N_ECDSA_P384_PKCS1_CERT_CHAIN, ecdsa_cert_chain_pem, S2N_MAX_TEST_PEM_SIZE));
+        EXPECT_SUCCESS(s2n_read_test_pem(S2N_ECDSA_P384_PKCS1_KEY, ecdsa_private_key_pem, S2N_MAX_TEST_PEM_SIZE));
+        EXPECT_SUCCESS(s2n_cert_chain_and_key_load_pem(rsa_cert, rsa_cert_chain_pem, rsa_private_key_pem));
+        EXPECT_SUCCESS(s2n_cert_chain_and_key_load_pem(ecdsa_cert, ecdsa_cert_chain_pem, ecdsa_private_key_pem));
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(server_config, rsa_cert));
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(server_config, ecdsa_cert));
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(server_config, "test_all"));
+
+        if (s2n_chacha20_poly1305.is_available()) {
+            /* Test chacha20 boosting when ciphersuites fail auth validation */
```

Comment:
Nit: Your test name might be overlooked here because of the spacing. Maybe add a newline?
```suggestion

        if (s2n_chacha20_poly1305.is_available()) {
        
            /* Test chacha20 boosting when ciphersuites fail auth validation */
```